src/write_outputs/energy_share_requirement/write_esr_prices.jl

```diff
@@ -19,6 +19,15 @@ function write_esr_prices(path::AbstractString, inputs::Dict, setup::Dict, EP::M
 	if setup["ParameterScale"] == 1
 		dfESR[!,:ESR_Price] = dfESR[!,:ESR_Price] * ModelScalingFactor # Converting MillionUS$/GWh to US$/MWh
 	end
-	CSV.write(joinpath(path, "ESR_prices.csv"), dfESR)
+
+	if haskey(inputs, "dfESR_slack")
+		dfESR[!,:ESR_AnnualSlack] = convert(Array{Float64}, value.(EP[:vESR_slack]))
+		dfESR[!,:ESR_AnnualPenalty] = convert(Array{Float64}, value.(EP[:eCESRSlack]))
+		if setup["ParameterScale"] == 1
```

Comment:
This function has two `if setup["ParameterScale"] == 1` blocks. They could be replaced using

`scale_factor = setup["ParameterScale"] == 1 ? ModelScalingFactor : 1`